src/cloudsc_pyiface/src/cloudscpytools/cloudsc_data.py

```diff
@@ -0,0 +1,457 @@
+    ]
+
+    argnames_buffer = [
+        'buffer_loc','buffer_tmp'
+    ]
+
+    argnames_tend = [
+        'tendency_loc_a','tendency_loc_t','tendency_loc_q',
+    ]
+
+    argnames_tend_cld = [
+        'tendency_loc_cld'
+    ]
+
+    argnames_nproma = [
+        'prainfrac_toprfz'
+    ]
+
+    for argname in argnames_nlev:
+        print('Define null field:',argname)
+        fields[argname] = np.zeros(shape=(nproma,nlev  ,nblocks), order='F')
+
+    for argname in argnames_nlevp:
+        print('Define null field:',argname)
+        fields[argname] = np.zeros(shape=(nproma,nlev+1,nblocks), order='F')
+
+    for argname in argnames_buffer:
+        print('Define null field:',argname)
+        fields[argname] = np.zeros(shape=(nproma,nlev,3+NCLV,nblocks), order='F')
+
+    for argname in argnames_tend:
+        print('Define null field:',argname)
+        fields[argname] = np.zeros(shape=(nproma,nlev,nblocks), order='F')
+
+    for argname in argnames_tend_cld:
+        print('Define null field:',argname)
+        fields[argname] = np.zeros(shape=(nproma,nlev,NCLV,nblocks), order='F')
+
+
+    for argname in argnames_nproma:
+        print('Define null field:',argname)
+        fields[argname] = np.zeros(shape=(nproma,nblocks), order='F')
+
+    fields['ydomcst']=clsc.yomcst.TOMCST()
+    fields['ydoethf']=clsc.yoethf.TOETHF()
+    fields['ydecldp']=clsc.yoecldp.TECLDP()
+    fields['ydephli']=clsc.yoephli.TEPHLI()
+
+    return fields
+
+def field_c_to_fortran(dims,cfield,nparms,klon):
+    """
+    field_c_to_fortran:
+    1) transposes C array input to Fortran array
+    2) rewrites Fortran linear array into block structure
+    """
+    ffieldtmp=np.asfortranarray(np.transpose(
+              np.ascontiguousarray(cfield)))
+    bfield=field_linear_to_block(dims,ffieldtmp,nparms,klon)
+    return bfield
+
+def field_linear_to_block(dims,lfield,nparms,nlon):
+    """
+    field_linear_to_block:
+    rewrites Fortran linear array into block structure
+    """
+    nproma =nparms['NPROMA']
+    nlev   =dims[1] #nparms['NLEV']
+    nblocks=nparms['NBLOCKS']
+    ngptot =nparms['NGPTOT']
+    ndim   =nparms['NDIM']
+    ldims=len(dims)
+    if lfield.dtype == "float64":
+        if ldims == 2:
+            b2field=np.asfortranarray(np.transpose(
+                                      np.zeros(shape=dims, dtype="float64")))
+            clsc.expand_mod.expand_r1(lfield, b2field,  nlon, nproma, ngptot, nblocks)
+            bfield=b2field
+        elif ldims == 3:
+            b3field=np.asfortranarray(np.transpose(np.zeros(shape=dims, dtype="float64")))
+            clsc.expand_mod.expand_r2(lfield, b3field,  nlon, nproma, nlev, ngptot, nblocks)
+            bfield=b3field
+        elif ldims == 4:
+            b4field=np.asfortranarray(np.transpose(np.zeros(shape=dims, dtype="float64")))
+#Note that we are using expand_r3bis here.
+#Original expand_r3 does not pass nlev from python to Fortran
+#            print ("This is 4D field")
```

Comment:
Debug leftover?

Reply:
No, rather an urgent note to discuss, because we don't want to carry on with expand_r3bis unless absolutely necessary. A very strange thing, on the Python side nlev was 137, on the Fortran side it was 5.